staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -666,6 +738,41 @@ func (p *sharedProcessor) resyncCheckPeriodChanged(resyncCheckPeriod time.Durati
 	}
 }
 
+func (p *sharedProcessor) removeListenerFor(handler ResourceEventHandler) {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	listeners := p.removeListenerLockedFor(handler)
+	if p.listenersStarted {
+		for _, l := range listeners {
+			close(l.addCh)
+		}
+	}
+}
+
+func (p *sharedProcessor) removeListenerLockedFor(handler ResourceEventHandler) []*processorListener {
+	if !reflect.ValueOf(handler).Type().Comparable() {
+		return nil
+	}
```

Comment:
this has been checked at line 630, does it need to be checked again?

Reply:
not really, for the actual call stack,  but the method might later on be called by other callers in a different context, and for the sake of defensive programming, I've added this to provide a safe function. 

But if you want I'll remove the check again.